src/Core/src/Hosting/MauiHandlersCollectionExtensions.cs

```diff
@@ -1,51 +1,34 @@
 using System;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.DependencyInjection.Extensions;
 
 namespace Microsoft.Maui.Hosting
 {
 	public static partial class MauiHandlersCollectionExtensions
 	{
-		public static IMauiHandlersCollection AddHandlers(this IMauiHandlersCollection handlersCollection, Dictionary<Type, Type> handlers)
-		{
-			foreach (var handler in handlers)
-			{
-				handlersCollection.AddTransient(handler.Key, handler.Value);
-			}
-			return handlersCollection;
-		}
-
-		public static IMauiHandlersCollection AddHandler(this IMauiHandlersCollection handlersCollection, Type viewType, Type handlerType)
+		public static IMauiHandlersCollection AddHandler(this IMauiHandlersCollection handlersCollection, Type viewType, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type handlerType)
 		{
 			handlersCollection.AddTransient(viewType, handlerType);
 			return handlersCollection;
 		}
 
-		public static IMauiHandlersCollection AddHandler<TType, TTypeRender>(this IMauiHandlersCollection handlersCollection)
-			where TType : IView
-			where TTypeRender : IViewHandler
+		public static IMauiHandlersCollection AddHandler<TType, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TTypeRender>(this IMauiHandlersCollection handlersCollection)
+			where TType : IElement
+			where TTypeRender : IElementHandler
```

Comment:
I loosened the type restriction here, so you can use this on Application and Window now?

Reply:
Thanks. This was probably around before we added IElementHandler.